streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -342,7 +342,18 @@ public void handleAssignment(final Map<TaskId, Set<TopicPartition>> activeTasks,
 
         maybeThrowTaskExceptions(taskCloseExceptions);
 
-        createNewTasks(activeTasksToCreate, standbyTasksToCreate);
+        final Collection<Task> newActiveTasks = createNewTasks(activeTasksToCreate, standbyTasksToCreate);
+        // If there are any transactions in flight and there are newly created active tasks, commit the tasks
+        // to avoid potential long restoration times.
+        if (processingMode == EXACTLY_ONCE_V2 && threadProducer().transactionInFlight() && !newActiveTasks.isEmpty()) {
+            log.info("New active tasks were added and there is an inflight transaction. Attempting to commit tasks.");
```

Comment:
One caveat for EOS-v2 is that, when we commit, we'd have to make sure we are committing all tasks that have processed any data, but not just the active tasks --- sorry for not making that clear before, since it also bothers me some time to make it straight, and as a result I filed https://issues.apache.org/jira/browse/KAFKA-14847, please feel free to read it in more details. --- in a word, when we are in EOS-v2, each commit has to include everyone even if we only want to commit for a part of that, so we'd better give all the tasks in the `commitTasksAndMaybeUpdateCommittableOffsets` func below.

Reply:
Oh I see. I read through the ticket and got some idea. Thanks for the explanation. I have updated the code to pass allTasks,

Reply:
Is INFO the right log level? Seem DEBUG might be more appropriate?

Reply:
I used INFO because I thought this is something users should be able to see by default. I am hoping this won't be a very frequent case and hence should be ok to notify by default.

Reply:
Ok. Not sure how important it is for users to see by default, but I agree it should not be too frequent as it should only happen during a rebalance.